src/samples/http/server-sse.c

```diff
@@ -0,0 +1,295 @@
+        /* this will loop trying to read as much data as possible to buffer. */
+        err = sol_util_load_file_fd_buffer(fd, &value);
+        if (err < 0) {
+            fprintf(stderr, "ERROR: failed to read from stdin: %s\n",
+                sol_util_strerrora(-err));
+            goto err;
+        }
+
+        if (value.used == 0) {
+            /* no data usually means ^D on the terminal, quit the application */
+            printf("no data on stdin, quitting.\n");
+            sol_quit();
+        } else {
+            err = sol_buffer_insert_slice(&value, 0, begin);
+            if (err < 0) {
+                fprintf(stderr, "ERROR: failed to append the data prefix: %s\n",
+                    sol_util_strerrora(-err));
+                goto err;
+            }
+
+            err = sol_buffer_append_slice(&value, end);
+            if (err < 0) {
+                fprintf(stderr, "ERROR: Could not append data to buffer: %s\n",
+                    sol_util_strerrora(-err));
+                goto err;
+            }
+        }
+    }
+
+    SOL_PTR_VECTOR_FOREACH_IDX (&responses, sse, i)
+        sol_http_response_progressive_feed(sse, sol_buffer_get_slice(&value));
+
+    return true;
+
+err:
+    sol_quit_with_code(EXIT_FAILURE);
+    stdin_watch = NULL;
+    return false;
+}
+
+static void
+delete_cb(void *data, const struct sol_http_response_progressive *sse)
+{
+    sol_ptr_vector_remove(&responses, sse);
+}
+
+static int
+request_events_cb(void *data, struct sol_http_request *request)
+{
+    int ret;
+    struct sol_http_response_progressive *sse;
+    struct sol_http_response response = {
+        SOL_SET_API_VERSION(.api_version = SOL_HTTP_RESPONSE_API_VERSION, )
+        .param = SOL_HTTP_REQUEST_PARAMS_INIT,
+        .response_code = SOL_HTTP_STATUS_OK,
+        .content = SOL_BUFFER_INIT_EMPTY
+    };
+
+    ret = sol_http_response_set_progressive_headers(&response);
+    if (ret < 0)
+        return ret;
+
+    sse = sol_http_server_send_response_progressive(request, &response, delete_cb, NULL);
+    sol_http_params_clear(&response.param);
+    if (!sse)
+        return -1;
+
+    ret = sol_ptr_vector_append(&responses, sse);
+    if (ret < 0) {
+        sol_http_response_progressive_del(sse, false);
+        return ret;
+    }
+
+    return 0;
+}
+
+static int
+request_cb(void *data, struct sol_http_request *request)
+{
+    int r;
+
+    SOL_BUFFER_DECLARE_STATIC(buf, SOL_INET_ADDR_STRLEN);
+    struct sol_network_link_addr addr;
+    struct sol_http_response response = {
+        SOL_SET_API_VERSION(.api_version = SOL_HTTP_RESPONSE_API_VERSION, )
+        .param = SOL_HTTP_REQUEST_PARAMS_INIT,
+        .response_code = SOL_HTTP_STATUS_OK,
+    };
+
+    r = sol_http_request_get_interface_address(request, &addr);
+    SOL_INT_CHECK(r, < 0, r);
+
+    SOL_NULL_CHECK(sol_network_link_addr_to_str(&addr, &buf), -1);
+
+    sol_buffer_init(&response.content);
+    r = sol_buffer_append_printf(&response.content, HTML_FILE,
```

Comment:
add a comment to say that SSE needs url to match so you need to do this. also test with a remote browser.